### PR TITLE
Correctly indent use items and functions

### PR DESCRIPTION
**Pull request by marcusklaas**
This should fix https://github.com/nrc/rustfmt/issues/41.

### CODE REVIEW

File: src/imports.rs [hunk 1]
```diff
@@ -29,7 +29,6 @@ impl<'a> FmtVisitor<'a> {
                             path_list: &[ast::PathListItem],
                             visibility: ast::Visibility,
                             vp_span: Span) -> String {
-        // FIXME check indentation
         let l_loc = self.codemap.lookup_char_pos(vp_span.lo);
```
Comment (nrc) [inline src/imports.rs]: This is still using the left hand side of the item to calculate the indent (and thus the space remaining) for the import. You need to pass in the indent from the caller of this method.

**Comment by nrc** [comment]
The test and changes to visitor.rs look good. I think you need to change imports.rs a little further as described in the inline comment.

**Comment by marcusklaas** [comment]
I made the changes, but it will be a bit harder now implement to fix https://github.com/nrc/rustfmt/issues/24, since `rewrite_use_list` only has a budget and doesn't know about `MAX_WIDTH` and `IDEAL_WIDTH`.

File: src/visitor.rs [hunk 2]
```diff
@@ -145,20 +145,24 @@ impl<'a, 'v> visit::Visitor<'v> for FmtVisitor<'a> {
...
+                        let block_indent = self.block_indent;
+                        let budget = IDEAL_WIDTH - block_indent;
+                        if let Some(new_str) = self.rewrite_use_list(block_indent,
+                                                                     budget,
+                                                                     path,
+                                                                     path_list,
+                                                                     item.vis) {
+                            self.changes.push_str_span(item.span, &new_str);
+                        }
                         self.last_pos = item.span.hi;
```
Comment (nrc) [inline src/visitor.rs]: I think you should only set last_pos if you push the string, otherwise we'll miss the import.

File: src/imports.rs [hunk 3]
```diff
@@ -89,10 +79,11 @@ impl<'a> FmtVisitor<'a> {
...
             }
         })).collect();
-        if path_str.len() == 0 {
+        let result = if path_str.len() == 0 {
             format!("{}use {{{}}};", vis, write_list(&items, &fmt))
         } else {
             format!("{}use {}::{{{}}};", vis, path_str, write_list(&items, &fmt))
-        }
+        };
+        Some(result)
```
Comment (nrc) [inline src/imports.rs]: This would be slightly better as `Some(if path_...})`

**Comment by marcusklaas** [comment]
Thanks for the feedback! I made the changes.

**Pull request merged**